modules/dists/BlockDist.chpl

```diff
@@ -1536,12 +1536,20 @@ where this.sparseLayoutType == unmanaged DefaultDist &&
 }
 
 // Block1 <=> Block2 
-proc BlockArr.doiSwap(arr) {
-  coforall (locarr1, locarr2) in zip(this.locArr, arr.locArr) {
-    on locarr1 {
-      locarr1.myElems <=> locarr2.myElems;
-      locarr1.locRAD <=> locarr2.locRAD;
+proc BlockArr.doiOptimizedSwap(other) {
+  if(this.dom.dist == other.dom.dist &&
+     this.dom.dist.dsiEqualDMaps(other.dom.dist)) {
+    coforall (locarr1, locarr2) in zip(this.locArr, other.locArr) {
+      on locarr1 {
+        locarr1.myElems.data <=> locarr2.myElems.data;
+        locarr1.myElems.initShiftedData();
+        locarr2.myElems.initShiftedData();
```

Comment:
Here, you are poking the internals of DefaultRectangular. I think you are doing that to avoid some of the shape computation, because that's checked already by dsiEqualDMaps above.

But from a design perspective, DefaultRectangular is a more integral part of the language definition than Block. In that sense, it is not a good idea to be playing with the internals of DefaultRectangular from Block, or any other module outside `modules/internals`.

That being said, I know that we have similar code in Block and maybe some other distributions. So, it is not completely off to add this, but if the performance overhead of that is small (I hope that it is), I'd still just swap myElems